plugins/module_utils/network/iosxr/config/bgp_global/bgp_global.py

```diff
@@ -0,0 +1,402 @@
+            'neighbor_log_message_in_value',
+            'neighbor_log_message_in_disable',
+            'neighbor_log_message_in_inheritance_disable',
+            'neighbor_log_message_out_value',
+            'neighbor_log_message_out_disable',
+            'neighbor_log_message_out_inheritance_disable',
+            'neighbor_update_in_filtering_attribute_filter_group',
+            'neighbor_update_in_filtering_logging_disable',
+            'neighbor_update_in_filtering_message_buffers',
+           ]
+
+        want_nbr = want.get("neighbors", {})
+        have_nbr = have.get("neighbors", {})
+        for name, entry in iteritems(want_nbr):
+            have = have_nbr.pop(name, {})
+            begin = len(self.commands)
+            self.compare(parsers=neighbor_parsers, want=entry, have=have)
+            neighbor_address = entry.get("neighbor", "")
+            if len(self.commands) != begin:
+                self.commands.insert(
+                    begin, self._tmplt.render({"neighbor": neighbor_address}, "neighbor", False)
+                )
+        for name, entry in iteritems(have_nbr):
+            if self._check_af("neighbor", name):
+                self._module.fail_json(
+                    msg="Neighbor {0} has address-family configurations. "
+                        "Please use the iosxr_bgp_neighbor_address_family module to remove those first.".format(
+                        name
+                    )
+                )
+            else:
+                self.addcmd(entry, "neighbor", True)
+
+    def _vrfs_compare(self, want, have):
+        """Custom handling of VRFs option
+        :params want: the want BGP dictionary
+        :params have: the have BGP dictionary
+        """
+        wvrfs = want.get("vrfs", {})
+        hvrfs = have.get("vrfs", {})
+        for name, entry in iteritems(wvrfs):
+            begin = len(self.commands)
+            vrf_have = hvrfs.pop(name, {})
+            self._compare_rpki_server(want=entry, have=vrf_have)
+            self._compare_neighbors(want=entry, have=vrf_have)
+            self.compare(parsers=self.parsers, want=entry, have=vrf_have)
+            if len(self.commands) != begin:
+                self.commands.insert(
+                    begin, self._tmplt.render({"vrf": entry.get('vrf')}, "vrf", False)
+                )
+        # cleanup remaining VRFs
+        # but do not negate it entirely
+        # instead remove only those attributes
+        # that this module manages
+        for name, entry in iteritems(hvrfs):
+            if self._check_af("vrf", name):
+                self._module.fail_json(
+                    msg="VRF {0} has address-family configurations. "
+                        "Please use the iosxr_bgp_address_family module to remove those first.".format(
+                        name
+                    )
+                )
+            else:
+                self.addcmd(entry, "vrf", True)
```

Comment:
I think that, the configs under have->vrf  (not in want), which do not belong to rpki_server, address_family and neighbors configurations,  are not handled here.